app/src/androidTest/java/tech/pauly/findapet/discover/DiscoverFragmentRobot.java

```diff
@@ -0,0 +1,70 @@
+package tech.pauly.findapet.discover;
+
+import android.content.Intent;
+import android.support.test.espresso.intent.rule.IntentsTestRule;
+
+import tech.pauly.findapet.R;
+import tech.pauly.findapet.dependencyinjection.EspressoPetApplication;
+import tech.pauly.findapet.shared.MainTabActivity;
+import tech.pauly.findapet.utils.MockWebServerRule;
+import tech.pauly.findapet.utils.RobotUtils;
+
+import static tech.pauly.findapet.utils.MockWebServerRequestPatterns.FETCH_ANIMALS;
+
+public interface DiscoverFragmentRobot {
```

Comment:
So Robot pattern is pretty simple, it's just a layer of abstraction from the test class to try and help make the tests more expressive, which makes them a lot more valuable. Hands: actions, Eyes: view assertions, Dependencies (not a fun name unfortunately) setting up anything that's not one of the others. Also get to trim down on repeated code for similar tests. It's a little bit bloated for tests as simple as this one, but I like the pattern in general.